pkg/config/generate_test.go

```diff
@@ -134,3 +134,40 @@ func testRequiredFields(cs *api.OpenShiftManagedCluster, t *testing.T) {
 	assert(c.NodeBootstrapKubeconfig != nil, "NodeBootstrapKubeconfig")
 	assert(c.AzureClusterReaderKubeconfig != nil, "AzureClusterReaderKubeconfig")
 }
+
+func TestInvalidateSecrets(t *testing.T) {
+	dpc := &api.PluginConfig{}
+	populate.Walk(dpc, nil)
+	cs := &api.OpenShiftManagedCluster{}
+	populate.Walk(cs, nil)
+
+	g := &simpleGenerator{
+		pluginConfig: *dpc,
+	}
+	saved := *cs
```

Comment:
needs to be `saved := cs.DeepCopy()` for 2 reasons:
1) makes intent very clear
2) takes care of copying all the pointer objects (more robust)